lib/bitex_bot/models/api_wrappers/bitex/bitex_api_wrapper.rb

```diff
@@ -0,0 +1,143 @@
+# Wrapper implementation for Bitex API.
+# https://bitex.la/developers
+class BitexApiWrapper < ApiWrapper
+  attr_accessor :api_key, :ssl_version, :debug, :sandbox
+
+  def initialize(settings)
+    self.api_key = settings.api_key
+    self.ssl_version = settings.ssl_version
+    self.debug = settings.debug
+    self.sandbox = settings.sandbox
+  end
+
+  def with_session
+    prev_key = Bitex.api_key
+    prev_sandbox = Bitex.sandbox
+    Bitex.api_key = api_key
+    Bitex.sandbox = sandbox
+    yield.tap do
+      Bitex.api_key = prev_key
+      Bitex.sandbox = prev_sandbox
+    end
+  end
+
+  def profile
+    with_session { Bitex::Profile.get }
+  end
+
+  def amount_and_quantity(order_id)
+    with_session do
+      closes = user_transactions.select { |t| t.order_id.to_s == order_id }
+      amount = closes.map { |c| c.usd.to_d }.sum.abs
+      quantity = closes.map { |c| c.btc.to_d }.sum.abs
+
+      [amount, quantity]
+    end
+  end
+
+  def balance
+    with_session { balance_summary_parser(profile) }
+  end
+
+  def find_lost(type, price, _quantity)
+    with_session { orders.find { |o| o.type == type && o.price == price && o.timestamp >= 5.minutes.ago.to_i } }
+  end
+
+  def order_book
+    with_session { order_book_parser(Bitex::MarketData.order_book) }
+  end
+
+  def orders
+    with_session { Bitex::Order.all.map { |o| order_parser(o) } }
+  end
+
+  def send_order(type, price, quantity)
+    { sell: Bitex::Ask, buy: Bitex::Bid }[type].create!(BitexBot::Settings.maker.order_book, quantity, price)
+  end
+
+  def transactions
+    with_session { Bitex::Trade.all.map { |t| transaction_parser(t) } }
+  end
+
+  def user_transactions
+    with_session { Bitex::Trade.all.map { |trade| user_transaction_parser(trade) } }
+  end
+
+  # {
+  #   usd_balance:               10000.00, # Total USD balance.
+  #   usd_reserved:               2000.00, # USD reserved in open orders.
+  #   usd_available:              8000.00, # USD available for trading.
+  #   btc_balance:            20.00000000, # Total BTC balance.
+  #   btc_reserved:            5.00000000, # BTC reserved in open orders.
+  #   btc_available:          15.00000000, # BTC available for trading.
+  #   fee:                            0.5, # Your trading fee (0.5 means 0.5%).
+  #   btc_deposit_address: "1XXXXXXXX..."  # Your BTC deposit address.
+  # }
+  def balance_summary_parser(balances)
+    BalanceSummary.new(
+      Balance.new(balances[:btc_balance], balances[:btc_reserved], balances[:btc_available]),
+      Balance.new(balances[:usd_balance], balances[:usd_reserved], balances[:usd_available]),
+      balances[:fee]
+    )
+  end
+
+  def last_order_by(price)
```

Comment:
It seems this method was copied over from ItBitApiWrapper but it's not needed in this one.